src/arrayops.jl

```diff
@@ -276,6 +280,10 @@ Base.@propagate_inbounds _pointer(arr::Base.FastContiguousSubArray, i, I) =
     pointer(arr, (i, I...))
 Base.@propagate_inbounds _pointer(arr::Base.FastContiguousSubArray, i, I::Tuple{}) =
     pointer(arr, i)
+Base.@propagate_inbounds _pointer(arr::Base.ReinterpretArray, i, I) =
+    pointer(arr, LinearIndices(arr)[i, I...])
+Base.@propagate_inbounds _pointer(arr::Base.ReinterpretArray, i, I::Tuple{}) =
+    pointer(arr, i)
```

Comment:
I think it'd be better to merge them with above methods using `Union{Array,Base.ReinterpretArray}` and `Union{Base.FastContiguousSubArray,Base.ReinterpretArray}`.